File: 524. Longest Word in Dictionary through Deleting.py
class Solution:
    def findLongestWord(self, s: str, d: List[str]) -> str:
        word = {i:w for i, w in enumerate(d)}
        ans, d = '', [list(w) for w in d]
        
        for c in s:
            for i in range(len(d)):
                if d[i] and d[i][0]==c:
                    d[i].pop(0)
                    
                    if not d[i] and (len(word[i])>len(ans) or\
                       (len(word[i])==len(ans) and word[i]<ans)):
                        ans = word[i]
        
        return ans

File: 524. Longest Word in Dictionary through Deleting.js
/**
 * @param {string} s
 * @param {string[]} d
 * @return {string}
 */
var findLongestWord = function(s, d) {
    let [word, ans] = [d.slice(0), ''];

    for(let i=0;i<d.length;i++){
        d[i] = d[i].split('');
    }
    for(const c of s){
        for([i, check] of Object.entries(d)){
            if(check.length>0 && check[0]==c){
                check.splice(0, 1);

                if (check.length==0 && (word[i].length>ans.length
                   ||(word[i].length==ans.length && word[i]<ans))){
                    ans = word[i];
                }
            }
        }
    }
    return ans;
};